packages/hoppscotch-common/src/services/persistence/__tests__/index.spec.ts

```diff
@@ -0,0 +1,2027 @@
+  setRESTCollections,
+} from "~/newstore/collections"
+import {
+  addGlobalEnvVariable,
+  environments$,
+  globalEnv$,
+  replaceEnvironments,
+  selectedEnvironmentIndex$,
+  setGlobalEnvVariables,
+  setSelectedEnvironmentIndex,
+} from "~/newstore/environments"
+import {
+  graphqlHistoryStore,
+  restHistoryStore,
+  setGraphqlHistoryEntries,
+  setRESTHistoryEntries,
+  translateToNewGQLHistory,
+  translateToNewRESTHistory,
+} from "~/newstore/history"
+import { bulkApplyLocalState, localStateStore } from "~/newstore/localstate"
+import {
+  applySetting,
+  bulkApplySettings,
+  performSettingsDataMigrations,
+  settingsStore,
+} from "~/newstore/settings"
+import { PersistenceService } from "../../persistence"
+import {
+  ENVIRONMENTS,
+  GQL_COLLECTIONS,
+  GQL_HISTORY,
+  GQL_TAB_STATE,
+  REST_COLLECTIONS,
+  REST_HISTORY,
+  REST_TAB_STATE,
+  VUEX_DATA,
+} from "./__mocks__"
+
+vi.mock("~/modules/i18n", () => {
+  return {
+    __esModule: true,
+    getI18n: vi.fn().mockImplementation(() => () => "test"),
+  }
+})
+
+// Define modules that are shared across methods here
+vi.mock("@vueuse/core", async (importOriginal) => {
+  const actualModule: Record<string, unknown> = await importOriginal()
+
+  return {
+    ...actualModule,
+    watchDebounced: vi.fn(),
+  }
+})
+
+vi.mock("~/newstore/environments", () => {
+  return {
+    addGlobalEnvVariable: vi.fn(),
+    setGlobalEnvVariables: vi.fn(),
+    replaceEnvironments: vi.fn(),
+    setSelectedEnvironmentIndex: vi.fn(),
+    environments$: {
+      subscribe: vi.fn(),
+    },
+    globalEnv$: {
+      subscribe: vi.fn(),
+    },
+    selectedEnvironmentIndex$: {
+      subscribe: vi.fn(),
+    },
+  }
+})
+
+/**
+ * Helper functions
+ */
+const spyOnGetItem = () => vi.spyOn(Storage.prototype, "getItem")
+const spyOnRemoveItem = () => vi.spyOn(Storage.prototype, "removeItem")
+const spyOnSetItem = () => vi.spyOn(Storage.prototype, "setItem")
+
+const getServiceInstance = () => {
+  const container = new TestContainer()
+  const service = container.bind(PersistenceService)
+  return service
+}
+
+describe("PersistenceService", () => {
+  afterAll(() => {
+    // Clear all mocks
+    vi.clearAllMocks()
+
+    // Restore the original implementation for any spied functions
+    vi.restoreAllMocks()
+  })
+
+  it.skip("`showErrorToast` method shows an error toast with the supplied key", () => {
+    vi.mock("~/composables/toast", () => {
+      return {
+        useToast: () => {
+          return {
+            error: vi.fn(),
+          }
+        },
+      }
+    })
+
+    const testKey = "testKey"
+    const toast = useToast()
+    const service = getServiceInstance()
+
+    // @ts-expect-error Testing private method
+    service.showErrorToast(testKey)
+
+    expect(toast.error).toHaveBeenCalledWith(
+      `There's a mismatch with the expected schema for the value corresponding to ${testKey} read from localStorage, keeping a backup in ${testKey}-backup`
+    )
+  })
```

Comment:
This test case is failing atm. Am I missing something wrt mocking? `showErrorToast` method -> [Ref](https://github.com/jamesgeorge007/hoppscotch/blob/refactor/persistence-service/packages/hoppscotch-common/src/services/persistence/index.ts#L90-L95).